code/modules/admin/view_variables/topic_list.dm

```diff
@@ -2,6 +2,17 @@
 /client/proc/vv_do_list(list/target, href_list)
 	var/target_index = text2num(GET_VV_VAR_TARGET)
 	if(check_rights(R_VAREDIT))
+		var/dmlist_varname = href_list["dmlist_varname"]
+		if(dmlist_varname)
+			var/dmlist_secure_level = GLOB.vv_special_lists[dmlist_varname]
+			if(dmlist_secure_level == VV_LIST_EDITABLE)
+				log_world("### vv_do_list() called: [src] attempted to edit a special list ([dmlist_varname]) Security-level:[dmlist_secure_level](allowed)")
+				log_admin("[key_name(src)] attempted to edit  a special list ([dmlist_varname]) Security-level:[dmlist_secure_level](allowed)")
+			else // fuck you exploiters
+				log_world("### vv_do_list() called: [src] attempted to edit a special list ([dmlist_varname]), but denied due to the Security-level:[dmlist_secure_level]")
+				log_admin("[key_name(src)] attempted to edit a special list ([dmlist_varname]), but denied due to the Security-level:[dmlist_secure_level]")
+				message_admins("[key_name_admin(src)] attempted to edit a special list ([dmlist_varname]), but denied due to the Security-level:[dmlist_secure_level]. Bonk this guy.")
+				return
```

Comment:
Being VV, there are a shit ton of security exploits in this entire system, but at the very least the security code itself shouldn't be vulnerable to HTML injections. Message_admins will hapilly execute scripts passed to it on the clients of other admins, so sanitise dmlist_varname as it is coming directly from users.

Reply:
HTML exploit is protected and okay already, but as you point out here, I figured this vulnerable from proc call. I will fix that

Reply:
added `IsAdminAdvancedProcCall()`. It works advanced proc call well.
href safety is a thing already.

Reply:
![image](https://github.com/user-attachments/assets/d61e7872-eb37-47c3-a7b5-cecb5c53e4fa)
![image](https://github.com/user-attachments/assets/f41187dc-8db9-41b7-97ef-6b681d1fc796)

Reply:
Whatever update you did missed the security exploit, the issue was with dmlist_varname which is direct user input being fed into message_admins not with the proc call. There isn't much we can do about proc calling.

Reply:
I am not sure what you're talking about. direct href injection through HTML is already protected.